kroxylicious-test-tools/src/main/java/io/kroxylicious/test/tester/KroxyliciousClients.java

```diff
@@ -64,14 +77,53 @@ public Consumer<String, String> consumer() {
 
     public <U, V> Consumer<U, V> consumer(Serde<U> keySerde, Serde<V> valueSerde, Map<String, Object> additionalConfig) {
         Map<String, Object> config = createConfigMap(bootstrapServers, additionalConfig);
-        return CloseableConsumer.wrap(new KafkaConsumer<>(config, keySerde.deserializer(), valueSerde.deserializer()));
+        Consumer<U, V> consumer = CloseableConsumer.wrap(new KafkaConsumer<>(config, keySerde.deserializer(), valueSerde.deserializer()));
+        consumers.add(consumer);
+        return consumer;
     }
 
     public KafkaClient simpleTestClient() {
         String[] hostPort = bootstrapServers.split(":");
         return new KafkaClient(hostPort[0], Integer.parseInt(hostPort[1]));
     }
 
+    public void close() {
+        List<Exception> exceptions = new ArrayList<>();
+        try {
+            for (Admin admin : admins) {
+                try {
+                    admin.close();
+                }
+                catch (Exception e) {
+                    exceptions.add(e);
+                }
+            }
```

Comment:
nit: personally I'd have extracted a method here. Something like

```java
Collection<Exceptions> batchClose(Collection<AutoCloseable> closeables) {
  List<Exceptions> exceptions = new ArrayList<>();
  for(AutoCloseable closeable : closeables) {
    try {
      closeable.close();
    } catch (Exception e) {
      exceptions.add(e);
    }
    return exceptions;
  }
  ```
  
And then called it for each of the collections.

Reply:
Tried it, unfortunately it's not possible to cast from `List<WhateverClass>` to `Collection<WhateverSuperClass>`, you get a big old error like this: `Inconvertible types; cannot cast 'java.util.List<org.apache.kafka.clients.admin.Admin>' to 'java.util.Collection<java.lang.AutoCloseable>'`.

Reply:
Does making it `Collection<? extends AutoCloseable>` help? If its just `Collection` vs `List` just use `List` everywhere...

Reply:
`Collection<? extends AutoCloseable>` fixed it 👍